File: 2107-find-unique-binary-string/2107-find-unique-binary-string.cpp
class Solution {
public:
    string res;
    bool backtrack(int n, string cur, const unordered_set<string> &s){
        if (cur.size() == n){
            if(s.find(cur) == s.end()){
                res = cur;
                return true;
            }
            return false;
        }

        for (char c: {'0','1'}){
            if (backtrack(n, cur + c, s))
                return true;
        }
        return false;
    }

    string findDifferentBinaryString(vector<string>& nums) {
        unordered_set<string> st(nums.begin(),nums.end());
        backtrack(nums.size(), "", st);

        return res;
    }
};

File: 2107-find-unique-binary-string/2107-find-unique-binary-string.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        st = set(nums)
        self.res = ""

        def backtrack(n, cur, s):
            if len(cur) == n:
                if cur not in s:
                    self.res = cur
                    return True
                return False

            for c in '01':
                if backtrack(n, cur + c, s):
                    return True

        backtrack(len(nums), "", st)
        return self.res